src/Disqus.php

```diff
@@ -11,105 +11,54 @@
-    /**
-     * @var integer
-     */
-    protected $debug;
-
-    protected $id;
-
-    /**
-     * @var array
-     */
-    protected $options = array(
+    const DISQUS_API_BASE_URI = 'https://disqus.com/api/3.0/';
+
+    private $httpClient;
+
+    private $apiKey;
+    private $secretKey;
+    private $debug;
+
+    private $shortname;
+
+    private $id;
+
+    private $options = [
         'since'   => null,
         'cursor'  => null,
         'query'   => null,
-        'include' => array('approved'),
+        'include' => ['approved'],
         'order'   => 'desc',
         'limit'   => 100,
         'debug'   => 0,
-    );
-
-    /**
-     * @param ContainerInterface $container
-     * @param string $apiKey
-     * @param string $secretKey
-     * @param string $baseUrl
-     * @param int    $debug
-     */
-    public function __construct(ContainerInterface $container, $apiKey, $secretKey = null, $baseUrl = null, $debug = 0)
+    ];
+
+    public function __construct(string $apiKey, ?string $secretKey, bool $debug = true)
     {
-        $this->container = $container;
+        $this->httpClient = HttpClient::createForBaseUri(self::DISQUS_API_BASE_URI);
 
         $this->apiKey    = $apiKey;
         $this->secretKey = $secretKey;
         $this->debug     = $debug;
-
-        if ($baseUrl) {
-            $this->baseUrl = $baseUrl;
-        }
     }
 
-    /**
-     * @param string $shortname
-     * @param array  $options
-     * @param string $fetch
-     *
-     * @return string
-     */
-    public function fetch($shortname, array $options, $fetch = 'threads/listPosts')
+    public function fetch(string $shortname, array $options, string $fetch = 'threads/listPosts'): array
     {
         $this->shortname = $shortname;
 
         $options = $this->setOptions($options);
 
         $url = $this->buildUrl($options, $fetch);
-        if ($this->container->has('knp_zend_cache.manager')) {
-            $cache = $this->container->get('knp_zend_cache.manager');
-            $cache = $cache->getCache($this->container->getParameter('knp_disqus.cache.'.$shortname));
-            $key   = sha1($url);
-            if (false === ($content = $cache->load($key))) {
-                $content = json_decode($this->httpRequest($url), true);
-
-                // we cache, even if we have a bad response
-                // sometimes disqus goes down (request times out), and we don't want to keep making
-                // this hanging request each time - just don't server-side load the comments for now
-                $cache->save($content, $key);
-            }
-        } else {
-            $content = json_decode($this->httpRequest($url), true);
-        }
+
+        $content = $this->request($url);
 
         // in case we got a bad response, fake some stuff
         if (!is_array($content) || !isset($content['response'])) {
-            $content = array('response' => false);
+            $content = ['response' => false];
```

Comment:
Let's throw an exception here. Maybe a `ClientException` from Symfony's HttpClient.

Reply:
I can't use `ClientException` because we don't have the response object here and `ClientException` requires it as argument. I did `RuntimeException` here because `ClientException` extends it.